javascript-carousel/main.js

```diff
@@ -0,0 +1,196 @@
+
+circleOne.addEventListener('click', clickToGoToOne);
+function clickToGoToOne(event) {
+  if (event.target && event.target.matches('.one')) {
+    clearInterval(intervalId);
+    intervalId = setInterval(goThruImages, 3000);
+    imageCount = 1;
+    image.setAttribute('src', 'images/025.png');
+    icon1.classList.replace('fa-regular', 'fa-solid');
+    icon5.classList.replace('fa-solid', 'fa-regular');
+    icon4.classList.replace('fa-solid', 'fa-regular');
+    icon3.classList.replace('fa-solid', 'fa-regular');
+    icon2.classList.replace('fa-solid', 'fa-regular');
+  }
+}
+
+circleTwo.addEventListener('click', clickToGoToTwo);
+function clickToGoToTwo(event) {
+  if (event.target && event.target.matches('.two')) {
+    clearInterval(intervalId);
+    intervalId = setInterval(goThruImages, 3000);
+    imageCount = 2;
+    image.setAttribute('src', 'images/001.png');
+    icon2.classList.replace('fa-regular', 'fa-solid');
+    icon5.classList.replace('fa-solid', 'fa-regular');
+    icon4.classList.replace('fa-solid', 'fa-regular');
+    icon3.classList.replace('fa-solid', 'fa-regular');
+    icon1.classList.replace('fa-solid', 'fa-regular');
+  }
+}
+
+circleThree.addEventListener('click', clickToGoToThree);
+function clickToGoToThree(event) {
+  if (event.target && event.target.matches('.three')) {
+    clearInterval(intervalId);
+    intervalId = setInterval(goThruImages, 3000);
+    imageCount = 3;
+    image.setAttribute('src', 'images/004.png');
+    icon3.classList.replace('fa-regular', 'fa-solid');
+    icon5.classList.replace('fa-solid', 'fa-regular');
+    icon4.classList.replace('fa-solid', 'fa-regular');
+    icon1.classList.replace('fa-solid', 'fa-regular');
+    icon2.classList.replace('fa-solid', 'fa-regular');
+  }
+}
+
+circleFour.addEventListener('click', clickToGoToFour);
+function clickToGoToFour(event) {
+  if (event.target && event.target.matches('.four')) {
+    clearInterval(intervalId);
+    intervalId = setInterval(goThruImages, 3000);
+    imageCount = 4;
+    image.setAttribute('src', 'images/007.png');
+    icon4.classList.replace('fa-regular', 'fa-solid');
+    icon5.classList.replace('fa-solid', 'fa-regular');
+    icon1.classList.replace('fa-solid', 'fa-regular');
+    icon3.classList.replace('fa-solid', 'fa-regular');
+    icon2.classList.replace('fa-solid', 'fa-regular');
+  }
+}
+
+circleFive.addEventListener('click', clickToGoToFive);
+function clickToGoToFive(event) {
+  if (event.target && event.target.matches('.five')) {
+    clearInterval(intervalId);
+    intervalId = setInterval(goThruImages, 3000);
+    imageCount = 5;
+    image.setAttribute('src', 'images/039.png');
+    icon5.classList.replace('fa-regular', 'fa-solid');
+    icon1.classList.replace('fa-solid', 'fa-regular');
+    icon4.classList.replace('fa-solid', 'fa-regular');
+    icon3.classList.replace('fa-solid', 'fa-regular');
+    icon2.classList.replace('fa-solid', 'fa-regular');
+  }
+}
```

Comment:
There's a lot of repetition here, you should try to find a way you can make a reusable function to accomplish this for all the dots.